tempodb/blocklist/poller.go

```diff
@@ -477,6 +477,13 @@ func (p *Poller) pollBlock(
 	var compactedBlockMeta *backend.CompactedBlockMeta
 
 	if !compacted {
+		noCompact, flagErr := p.reader.HasNoCompactFlag(derivedCtx, blockID, tenantID)
+		if flagErr != nil {
+			return nil, nil, fmt.Errorf("failed to check nocompact flag: %w", flagErr)
+		}
+		if noCompact {
+			return nil, nil, nil
+		}
```

Comment:
Blocks can't be skipped here. This is called for building the index, which would exclude them in the read path.

I think the blocklist needs to either return that info when `List.Metas()` is called or to conditionally include/exclude them.